src/core/utils/useReaderPlayer.tsx

```diff
@@ -1,51 +1,81 @@
-import { useGetV1UserLoans } from "../publizon/publizon";
 import { Manifestation } from "./types/entities";
 import { getManifestationIsbn } from "../../apps/material/helper";
 import {
   getOrderIdByIdentifier,
-  getReaderPlayerType
+  getReaderPlayerType,
+  getReservedReservation
 } from "../../components/reader-player/helper";
-import { mapPublizonLoanToLoanType } from "./helpers/list-mapper";
 import { isAnonymous } from "./helpers/user";
+import useOnlineAvailabilityData from "../../components/availability-label/useOnlineAvailabilityData";
+import { hasCorrectAccess } from "../../components/material/material-buttons/helper";
+import {
+  useGetV1UserLoans,
+  useGetV1UserReservations
+} from "../publizon/publizon";
+import {
+  mapPublizonLoanToLoanType,
+  mapPublizonReservationToReservationType
+} from "./helpers/list-mapper";
 
 const useReaderPlayer = (manifestations: Manifestation[] | null) => {
   const isUserAnonymous = isAnonymous();
-  const { data } = useGetV1UserLoans(
+  const hasManifestations = !!manifestations?.length;
+
+  const type =
+    hasManifestations && hasCorrectAccess("Ereol", manifestations)
+      ? getReaderPlayerType(manifestations)
```

Comment:
Is there a reason why the checks `hasManifestations && hasCorrectAccess("Ereol", manifestations)` is not embedded within `getReaderPlayerType()`?

If we can only determine the type if these exist then it seems reasonable to have them embedded in the check.